libsolutil/SetOnce.h

```diff
@@ -79,6 +79,8 @@ class SetOnce
 	/// @throws BadSetOnceAccess when the stored value has not yet been set
 	T const* operator->() const { return std::addressof(**this); }
 
+	/// @return true if a value was assigned
+	bool set() const { return m_value.has_value(); }
```

Comment:
I think `hasValue()` or `isSet()` would be a less ambiguous name.

```suggestion
	bool hasValue() const { return m_value.has_value(); }
```

And by the way, maybe adding a `value()` function would be a good idea too?

Reply:
I used `isSet()` before and then @chriseth wanted it to be `set()` :laughing:

Reply:
Oh :) Yeah, names can be pretty subjective. I'd personally vote for `hasValue()` but I remember that @chriseth prefers a convention if there's no verb in the accessor so `set()` is probably more consistent with the rest of the code.

Reply:
or we use the same that `optional` has and use `has_value` that way `SetOnce` and `optional` can be used in a more similar way, avoiding confusion.

Reply:
Using a different convention for our stuff has the nice side-effect of making it easy to see what was added by us and what's built in. Not sure if that matters much in this simple case, but if I saw `has_value()`, I'd assume that it's probably `optional`, not `SetOnce`.

Reply:
It's mostly about `is` being redundant in functions returning bool.